### PR TITLE
feat: hardened images

### PR DESCRIPTION
**Pull request by gduncklee**
Adds hardening bits and pieces for our suite of images.

### CODE REVIEW

File: images/Dockerfile-python
Comment (jcadam14) [inline images/Dockerfile-python:1]: Do these changes need to be applied across the board for any Dockerfile built from python alpine?

**Comment by jcadam14** [comment]
I've checked out all the python based images and they all work fine.  Looking at https://github.com/cfpb/sbl-frontend/pull/530 it seems the node and nginx images are also good (though that PR needs a main merge).  So I think this should be good to merge once that frontend PR is merged in?

**Review by jcadam14 — DISMISSED** [review]
Everything seems good

**Review by hkeeler — DISMISSED** [review]

**Review by hkeeler — APPROVED** [review]

**Pull request merged**
